packages/autorest.python/test/vanilla/legacy/Expected/AcceptanceTests/XmsErrorResponse/xmserrorresponse/aio/operations/_pet_operations.py

```diff
@@ -99,8 +91,15 @@ async def get_pet_by_id(self, pet_id: str, **kwargs: Any) -> Optional[_models.Pe
         response = pipeline_response.http_response
 
         if response.status_code not in [200, 202]:
-            map_error(status_code=response.status_code, response=response, error_map=error_map)  # type: ignore
-            raise HttpResponseError(response=response)
+            map_error(status_code=response.status_code, response=response, error_map=error_map)
+            error = None
+            if response.status_code == 400:
+                error = self._deserialize.failsafe_deserialize(str, pipeline_response)
+            elif response.status_code == 404:
+                error = self._deserialize.failsafe_deserialize(_models.NotFoundErrorBase, pipeline_response)
```

Comment:
```suggestion
                error = self._deserialize.failsafe_deserialize(_models.NotFoundErrorBase, response)
```
I think it shall be `response` instead of `pipeline_response` here.